src/store-service.js

```diff
@@ -90,12 +90,24 @@
         }
 
         closeWindow() {
+            var closedArray;
             this.store.closed = Date.now();
             this.save();
-            this.$rootScope.$broadcast('closeWindow');
+
+            if (this.store.stocks.length > 0) {
+                this.$rootScope.$broadcast('closeWindow');
+            }
+
+            // Trim any stores without stocks
+            closedArray = storage.filter((store) => store.closed !== 0);
+            var emptyStores = closedArray.filter((store) => store.stocks.length === 0);
+            emptyStores.forEach((emptyStore) => {
+                var index = storage.indexOf(emptyStore);
+                storage.splice(index, 1);
+            });
 
             // Trim the oldest closed store
-            var closedArray = storage.filter((store) => store.closed !== 0);
+            closedArray = storage.filter((store) => store.closed !== 0);
```

Comment:
Instead of re-setting `closedArray`, maybe it's worth doing something like:

```js
restorableWindows = storage.filter((store) => store.closed !== 0 && store.stocks.length > 0);
var cleanableWindows = storage.filter((store) => store.closed !== 0 && store.stocks.length === 0);
```

I'm not sure that's much better. It's still 2 iterations through `storage`